rfc/rfc-0011-streaming-gateways.md

```diff
@@ -0,0 +1,68 @@
+# RFC-0011: Streaming Gateways
+
+**Authors:** @scothis, @markfisher, @ericbottard
+
+**Status:**
+
+**Pull Request URL:** [#1376](https://github.com/projectriff/riff/pull/1376)
+
+**Superseded by:** N/A
+
+**Supersedes:** N/A
+
+**Related:** N/A
+
+
+## Problem
+
+The relationships between a `Stream` and a `(InMemory|Kafka|Pulsar)Provider` is fragile. The stream contains a hostname for the provider's provisioner, which the stream controller calls when reconciling the stream. This has a few problems:
+- the `Stream` resource doesn't know when the `Provider` is unavailable, or deleted
+- the `Stream` has no way automatically obtain a new address for the provisioner
+- the provisioner name is completing in the `Service` namespace. The likelihood of a name collision is reduced by appending a suffix to the service name, however, the user must include this suffix when creating each stream
+- as new providers are created, there is a significant amount of reconciliation logic that is duplicated with trivial differences
+
+The problem is derived from [projectriff/system#86](https://github.com/projectriff/system/issues/86).
+
+### Anti-Goals
+
+This RFC will not prevent naming collisions between different provisioners with the same name. The first gateway of a given name wins.
```

Comment:
```suggestion
This RFC will not prevent naming collisions between different provisioners using the same stream name. The first gateway trying a given name wins.
```

Reply:
this is talking about gateway names, not stream names

Reply:
Ah, of course.